Content.Shared/_Moffstation/Overlay/Components/NightVisionComponent.cs

```diff
@@ -1,10 +1,21 @@
 using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
 
 namespace Content.Shared._Moffstation.Overlay.Components;
 
-/// <summary>
-/// When this component is on a player-controlled entity, it applies a night vision visual effect and creates a client-
-/// only pointlight to illuminate the area around the entity.
-/// </summary>
-[RegisterComponent, NetworkedComponent]
-public sealed partial class NightVisionComponent : Component;
+/// When this component is on a player-controlled entity, it applies a night vision visual effect
+/// (<see cref="ShaderPrototype"/>) and creates a <see cref="EffectPrototype">client-only effect entity</see> with a
+/// pointlight to illuminate the area around the entity.
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class NightVisionComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public EntProtoId EffectPrototype = "EffectNightVision";
+
+    [DataField, AutoNetworkedField]
+    public EntProtoId ShaderPrototype = "ModernNightVisionShader";
```

Comment:
hunch, might need to be of form
```suggestion
    public ProtoId<ShaderPrototype> ShaderPrototype = "ModernNightVisionShader";
```

Reply:
bleh. Can't do it because it's a shared component and ShaderProto is client only.

Reply:
I saw one other overlay hardcode the shader proto in itself, so that's what I did.

Reply:
yea thats fine then